dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -394,4 +395,36 @@ def _convert_time_dimension(self, load_data_df, project_config):
 
         if time_dim.model.is_time_zone_required_in_geography():
             load_data_df = load_data_df.drop("time_zone")
+
+        load_data_df = self._convert_dataset_time_to_project_time_interval(
+            load_data_df, project_config
+        )
```

Comment:
Should this be part of time_dim.convert_dataframe instead of a post-process?

I'm asking in part because reading _convert_dataset_time_to_project_time_interval is confusing. The input load_data_df has already been partially converted to project time, but not completely, and just looking at that function it is not clear that that should be the case.